### PR TITLE
First pass at adding JavaDocs for new java api classes

### PR DESCRIPTION
**Pull request by andrewfayres**
## Description ##
Adding in Javadocs for all the new java api classes. I looked at the resulting docs after rebuilding and it seems to look as expected. Most of these are the same as the docs in the Scala code. I removed the references to default values and made other changes for places that I saw them.

@piyushghai @lanking520 @nswamy 

### Essentials ###
Please feel free to remove inapplicable items for your PR.
- [ ] The PR title starts with [MXNET-$JIRA_ID], where $JIRA_ID refers to the relevant [JIRA issue](https://issues.apache.org/jira/projects/MXNET/issues) created (except PRs with tiny changes)
- [ ] Changes are complete (i.e. I finished coding on this PR)
- [ ] All changes have test coverage:
- Unit tests are added for small changes to verify correctness (e.g. adding a new operator)
- Nightly tests are added for complicated/long-running ones (e.g. changing distributed kvstore)
- Build tests will be added for build configuration changes (e.g. adding a new build option with NCCL)
- [ ] Code is well-documented: 
- For user-facing API changes, API doc string has been updated. 
- For new C++ functions in header files, their functionalities and arguments are documented. 
- For new examples, README.md is added to explain the what the example does, the source of the dataset, expected performance on test set and reference to the original paper if applicable
- Check the API doc at http://mxnet-ci-doc.s3-accelerate.dualstack.amazonaws.com/PR-$PR_ID/$BUILD_ID/index.html
- [ ] To the my best knowledge, examples are either not affected by this change, or have been fixed to be compatible with this change

### Changes ###
- [ ] Feature1, tests, (and when applicable, API doc)
- [ ] Feature2, tests, (and when applicable, API doc)

## Comments ##
- If this change is a backward incompatible change, why must this change be made.
- Interesting edge cases to note here

### CODE REVIEW

**Comment by roywei** [comment]
@andrewfayres Thanks for the contribution!
@mxnet-label-bot [pr-awaiting-review]

**Comment by roywei** [comment]
@mxnet-label-bot [pr-awaiting-merge,Java]

File: scala-package/core/src/main/scala/org/apache/mxnet/javaapi/Context.scala [hunk 1]
```diff
@@ -18,6 +18,12 @@ package org.apache.mxnet.javaapi
 
 import collection.JavaConverters._
 
+/**
+  * Constructing a context.
```
Comment (nswamy) [inline scala-package/core/src/main/scala/org/apache/mxnet/javaapi/Context.scala]: Can you briefly describe what is a context?

File: scala-package/core/src/main/scala/org/apache/mxnet/javaapi/NDArray.scala [hunk 2]
```diff
@@ -65,37 +102,99 @@ object NDArray {
...
+    */
   def arange(start: Float, stop: Float, step: Float, repeat: Int,
              ctx: Context, dType: DType.DType): NDArray =
     org.apache.mxnet.NDArray.arange(start, Some(stop), step, repeat, ctx, dType)
 }
 
+/**
+  * NDArray object in mxnet.
+  * NDArray is basic ndarray/Tensor like data structure in mxnet. <br />
+  * <b>
```
Comment (nswamy) [inline scala-package/core/src/main/scala/org/apache/mxnet/javaapi/NDArray.scala]: Or use it inside a try-with-resources() or user [[ResourceScope]] in a try-with-resources block

File: scala-package/core/src/main/scala/org/apache/mxnet/javaapi/NDArray.scala [hunk 3]
```diff
@@ -65,37 +102,99 @@ object NDArray {
...
+  * WARNING: it is your responsibility to clear this object through dispose().
```
Comment (nswamy) [inline scala-package/core/src/main/scala/org/apache/mxnet/javaapi/NDArray.scala]: does this come out as WARNING or a Note when you see the JavaDocs, if not can you follow the JavaDocs standard of adding note wherever you add WARNINGs or Notes
https://stackoverflow.com/questions/5619575/javadoc-where-to-add-notes-remarks-to-documentation

File: scala-package/core/src/main/scala/org/apache/mxnet/javaapi/NDArray.scala [hunk 4]
```diff
@@ -108,28 +207,88 @@ class NDArray(val nd : org.apache.mxnet.NDArray ) {
 
   def serialize() : Array[Byte] = nd.serialize()
 
+  /**
+    * Release the native memory. <br />
+    * The NDArrays it depends on will NOT be disposed. <br />
+    * The object shall never be used after it is disposed.
+    */
   def dispose() : Unit = nd.dispose()
```
Comment (nswamy) [inline scala-package/core/src/main/scala/org/apache/mxnet/javaapi/NDArray.scala]: I am contemplating if we ever need to expose this at all or should we direct users to use it along with ResourceScope to make the user experience better

File: scala-package/infer/src/main/scala/org/apache/mxnet/infer/javaapi/ObjectDetector.scala [hunk 5]
```diff
@@ -26,7 +26,20 @@ import org.apache.mxnet.javaapi.{Context, DataDesc, NDArray}
 import scala.collection.JavaConverters
 import scala.collection.JavaConverters._
 
-
+/**
+  * A class for object detection tasks
```
Comment (nswamy) [inline scala-package/infer/src/main/scala/org/apache/mxnet/infer/javaapi/ObjectDetector.scala]: Would it make sense add a brief about what ObjectDetector is? such as 
```This class helps to run ObjectDetection tasks where given a image it finds bounding boxes and corresponding labels```

File: scala-package/infer/src/main/scala/org/apache/mxnet/infer/javaapi/Predictor.scala [hunk 6]
```diff
@@ -22,6 +22,21 @@ import org.apache.mxnet.javaapi.{Context, DataDesc, NDArray}
 import scala.collection.JavaConverters
 import scala.collection.JavaConverters._
 
+/**
+  * Implementation of prediction routines.
```
Comment (nswamy) [inline scala-package/infer/src/main/scala/org/apache/mxnet/infer/javaapi/Predictor.scala]: same as above? do you think we should brief write what predictor is?

File: scala-package/core/src/main/scala/org/apache/mxnet/javaapi/NDArray.scala (hunk 3 above)
Comment (andrewfayres) [reply scala-package/core/src/main/scala/org/apache/mxnet/javaapi/NDArray.scala]: The warning comes out looking like this: https://mxnet.incubator.apache.org/api/scala/docs/index.html#org.apache.mxnet.NDArray

**Comment by nswamy** [comment]
these are really small changes, lets just do it now. Agree they apply to Scala as well, you can create a JIRA for the Scala API doc update.

**Comment by andrewfayres** [comment]
Updated them and made a [JIRA task](https://issues.apache.org/jira/browse/MXNET-1178) for updating the ScalaDocs.

**Pull request merged**